javascript/channel.ts

```diff
@@ -50,40 +54,104 @@ export class Channel {
                     resolve(undefined);
                     return;
                 }
-                if (this.readBuf.length >= len) {
```

Comment:
@progrium: in the Go implementation IIUC this will return any available data from the buffer immediately, but in the TS implementation it's waiting until it can fill the `len` bytes requested. Was there a reason for that difference?
I updated this to match what the Go code appears to be doing since that also made it easier to handle the window adjustments after reading data.

Reply:
I think it might have been human error! There might have been something that led to this but I'd have to go back in look. When in doubt, trust the go